src/main/java/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,24 @@
+package strategy;
+
+import java.util.HashMap;
+import java.util.Map;
+import model.Fruit;
+import service.OperationHandler;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<String, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl() {
+        this.operationHandlerMap = new HashMap<>();
+    }
+
+    @Override
+    public void addOperation(String type, OperationHandler operation) {
+        operationHandlerMap.put(type, operation);
+    }

```

Comment:
```suggestion
    public OperationStrategyImpl(Map<String, OperationHandler> operationHandlerMap) {
        this.operationHandlerMap = operationHandlerMap;
    }

```
Давай мапу створимо ззовні, і передамо її сюли уже готову